OpenSim/Simulation/Model/PrescribedForce.cpp

```diff
@@ -235,13 +256,15 @@ void PrescribedForce::computeForce(const SimTK::State& state,
     const bool hasPointFunctions  = pointFunctions.getSize()==3;
     const bool hasTorqueFunctions = torqueFunctions.getSize()==3;
 
-    assert(_body!=0);
+    const PhysicalFrame& frame =
+        getConnector<PhysicalFrame>("frame").getConnectee();
+
     if (hasForceFunctions) {
         Vec3 force(forceFunctions[0].calcValue(timeAsVector), 
                    forceFunctions[1].calcValue(timeAsVector), 
                    forceFunctions[2].calcValue(timeAsVector));
         if (!forceIsGlobal)
-            engine.transform(state, *_body,                 force, 
+            engine.transform(state, frame,                 force,
                                     getModel().getGround(), force);
```

Comment:
Could use the `Frame` itself to re-express the force in Ground. It would be much more readable and would start to remove our dependencies on the `SimbodyEngine`. Same is true for the transformations below that depend on the `engine`.